blocks/carousel/carousel.js

```diff
@@ -36,16 +41,13 @@ export default function decorate(block) {
   optimizeThumbnails(block);
   addEnclosingDiv(block);
 
-  const sliderWrapper = createSliderWrapper(block.firstElementChild.lastElementChild);
-  const slider = createPictureSlider();
-  const slides = setSliderIds();
-  if (slides.length > 1) {
-    const arrowNav = createArrowNav();
-    slider.appendChild(arrowNav);
-
-    const dottedNavContainer = createDottedNav();
-    sliderWrapper.appendChild(dottedNavContainer);
+  const carousel = new Carousel(block.firstElementChild.lastElementChild);
+  carousel.createPictureSlider();
+  carousel.setSliderIds();
 
-    initSlider();
+  if (carousel.sliderIds.length > 1) {
```

Comment:
this business logic should be moved to the carousel with a method like `carousel.hasSlides()` or similar.